api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,486 @@
+            Maximum time in seconds to wait for the flushing of each chunk; if the `batch_limit` is reached before this
+            time is reached the messages are immediately flushed to the client.
+            A value of 0 will flush every batch events immediatly as soon as they come in.
+          type: integer
+          format: int32
+          required: false
+        - name: stream_timeout
+          in: query
+          description: |
+            Maximum time in seconds a stream will live before being interrupted. If unspecified will stream
+            indefinately.
+            If this timeout is reached any pending messages (in the sense of `stream_limit`) will be flushed to the
+            client.
+            Stream initialization will fail if `stream_timeout` is lower than `batch_flush_timeout`.
+          type: integer
+          format: int32
+          required: false
+        - name: batch_keep_alive_limit
+          in: query
+          description: |
+            Maximum number of keep-alive messages to get in a row
+            before closing the connection.
+            Unlimited by default.
+          type: integer
+          format: int32
+          required: false
+        - name: Accept-Encoding
+          in: header
+          required: false
+          description: '"If set the client requests the body to be gzip compressed. For Example "Content-Encoding: gzip"'
+          type: string
+      responses:
+        '200':
+          description: |
+            Starts streaming to the client.  
+            Stream format is a continuous series of
+            `SimpleStreamEvent`s separated by `\n`
+          schema:
+            $ref: '#/definitions/SimpleStreamEvent'
+        '400':
+          description: Bad syntax
+          schema:
+            $ref: '#/definitions/Problem'
+        '401':
+          description: Not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+        '404':
+          description: Not found. Typically in case of requesting from a non existing topic of partition.
+          schema:
+            $ref: '#/definitions/Problem'
+        '500':
+          description: Not found. Typically in case of requesting from a non existing topic of partition.
+          schema:
+            $ref: '#/definitions/Problem'
+
+
+# ################################### #
+#                                     #
+#             Definitions             #
+#                                     #
+# ################################### #
+
+definitions:
+  Event:
+    type: object
+    description: |
+      This is the most general representation of an event, that can be processed
+      by Nakadi.
+
+      It should be used as a base definition for all events, that flow through
+      Nakadi by extending attributes of this object type.
+    properties:
+      event_type:
+        type: string
+        example: 'https://resource-events.zalando.com/ResourceCreated'
+      partitioning_key:
```

Comment:
For me a 'key' is an attribute by which you can retrieve or identify a data record, i.e. event instance here.
The partitioning_field here is used as explicit input for computation of the event partition (and not as key in the sense above).

Additional hints wrt. partitioning here: there are 2 levels of client control:
a) control by passing the partitioning criteria:
Here Nakadi maps it to event partition range using a specific hash with low collision probability. 
b) full control by passing the partition value:
Here Nakadi directly uses the partition value and refuses out of range values. 
 
I think Nakadi should support both variants. 
Variant a) is covered by current proposal -- P.S: we should disclose here also the used hash function.
Variant b) should be covered by new partitionId input param.

P.S: If not explicitly provided, Nakadi will use a default partition computation which ensures equally distribution among available event topic partitions.